Earthfile

```diff
@@ -79,3 +79,20 @@ fmt:
 check-dependencies:
     FROM +copy-src
     DO lib-rust+CARGO --args="deny --all-features check --deny warnings bans license sources"
+
+# publish publishes the target crate to cargo.io. Must specify package by --PACKAGE=<package-name>
+publish:
+    FROM +copy-src
+    ARG --required PACKAGE
+    ARG DRY_RUN=true
+
+    # earthly doesn't support pass secrets to FUNCTION directly, copy it to env through a file
+    RUN --secret CARGO_REGISTRY_TOKEN echo $CARGO_REGISTRY_TOKEN > ~/.CARGO_REGISTRY_TOKEN
+    ENV CARGO_REGISTRY_TOKEN=$(cat ~/.CARGO_REGISTRY_TOKEN)
```

Comment:
https://docs.earthly.dev/docs/earthfile#env-same-as-dockerfile-env says:
> Do not use the ENV command for secrets used during the build. All ENV values used during the build are persisted within the image itself

Maybe it's ok here since we aren't actually making a final image.

But `DO lib-rust+CARGO` is mainly about caching the `target` dir for incremental builds -- is that useful `cargo publish`? I suspect not so maybe we can just do `RUN --secret cargo publish...`?

Reply:
I think in this scenario we care more about protecting secrets than caching. I agree we should switch to `RUN --secret cargo publish...` 👍